packages/autorest.python/autorest/codegen/models/code_model.py

```diff
@@ -77,6 +77,19 @@ def __init__(
             t for t in self.types_map.values() if isinstance(t, CombinedType) and t.name
         ]
 
+    @property
+    def has_form_data(self) -> bool:
+        for client in self.clients:
```

Comment:
this should be a property on the body parameter, that we can bubble up. I.e. the on the body parameter we should have `@property def is_form_data(self) -> bool`, which we can then bubble up as `@property def has_form_data_body` onto the operation, which we can then look for in `code_model`. We really want to tie information for a property onto itself

Reply:
make sense.